torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -178,6 +179,10 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
         chunks = _chunk_with_empty(param_data, shard_world_size, dim=0)
         sharded_param = chunks[shard_rank]
         self.sharded_size = sharded_param.size()
+        # TODO: DTensor's local tensor may not have the correct stride for the
+        # uneven sharding case, so we construct a new tensor to get the stride:
+        # https://github.com/pytorch/pytorch/issues/118836
+        self.sharded_stride = torch.empty(self.sharded_size).stride()
```

Comment:
we just want to compute the contiguous strides of a particular shape?  cant we do it less cheaply than allocating a tensor- perhaps we have helpers for this but only in c++ and no bindings? probably fine to leave like this.  (only happens on init path so nbd i guess)

Reply:
fwiw there's this:

```python
>>> torch._prims_common.make_contiguous_strides_for(torch.Size([20, 10, 5]))
(50, 5, 1)
```

Reply:
Good suggestions! For clarity, I renamed the variable to `self.contiguous_sharded_stride` and used `make_contiguous_strides_for()` (still at initialization time).